apiserver/charms.go

```diff
@@ -532,7 +538,17 @@ func (h *charmsHandler) processGet(r *http.Request, st *state.State) (
 // the error is encoded in the Error field as a string, not an Error
 // object.
 func sendJSONError(w http.ResponseWriter, req *http.Request, err error) error {
-	logger.Errorf("returning error from %s %s: %s", req.Method, req.URL, errors.Details(err))
+	if errors.IsNotYetAvailable(err) {
+		// This error is typically raised when trying to fetch the blob
+		// contents for a charm which is still pending to be downloaded.
+		//
+		// We should log this at debug level to avoid unnecessary noise
+		// in the logs.
+		logger.Debugf("returning error from %s %s: %s", req.Method, req.URL, errors.Details(err))
```

Comment:
Can we have a better message here...

`charm currently not available, returning error from ...`

Reply:
I intentionally kept it the same as the error above it for consistency; you will also get this information in the logged error details.